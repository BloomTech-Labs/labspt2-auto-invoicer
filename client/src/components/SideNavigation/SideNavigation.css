.navigation {
  position: fixed;
  width: 100%;
  z-index: 1;
  margin: 5px 0 5px 10px;
}

.navbar {
  display: flex;
  align-items: center;
  justify-content: space-between;
}
#drawer-container {
  min-width: 180px;
  margin: 10px 0 5px 10px;
}
/* Icon Cards in Side Nav */
.navbar button:nth-child(1):hover {
  background-color: none;
}
#hamburger {
  padding: 5px;
  font-size: 24px;
}
#left-icon {
  padding: 5px;
  font-size: 24px;
}
.all-icon-container {
  width: 100%;
}

.icon-container {
  text-decoration: none;
  color: black;
}
.icon-container .icon-item:hover {
  background-color: lightgray;
}
.icon-title {
  font-size: 2.4rem;
}

.active .icon-item {
  background-color: gray;
  color: blue;
}
.active .icon-item:hover {
  background-color: gray;
  color: blue;
}

.auth-container {
  display: flex;
  width: 300px;
  justify-content: space-evenly;
  align-items: center;
}
.authentication-btns {
  font-family: 'Roboto', sans-serif;
  font-size: 1.6rem;
  border: 1px solid gray;
  border-radius: 5px;
  padding: 5px 10px;
  width: 90px;
}

.authentication-btns:hover {
  border: 1px solid darkgray;
  background: gray;
  color: white;
  box-shadow: 3px 3px 3px lightgray;
}

#signOut {
  font-size: 1.8rem;
  border: 1px solid gray;
  border-radius: 5px;
  padding: 5px 10px;
  text-decoration: none;
  color: inherit;
}
#signOut:hover {
  border: 1px solid darkgray;
  background: gray;
  color: white;
  box-shadow: 3px 3px 3px lightgray;
}

.credits {
  font-size: 1.6rem;
}

@media (max-width: 500px) {
  .navigation {
    width: 98%;
    margin: 0 auto;
  }
  .all-icon-container {
    max-width: 100%;
  }
  .all-icon-container .icon-container li {
    display: flex;
    justify-content: center;
  }
  .auth-container {
    justify-content: flex-end;
  }
  .authentication-btns {
    margin-left: 5px;
    width: auto;
  }
  #signOut {
    margin-left: 10px;
  }
  #drawer-container {
    margin: 5px 0 0 10px;
  }
}
