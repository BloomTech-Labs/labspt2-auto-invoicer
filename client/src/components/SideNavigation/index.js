import React from "react";
// list of material ui components
import Drawer from "@material-ui/core/Drawer";
import List from "@material-ui/core/List";
import Divider from "@material-ui/core/Divider";
import ListItem from "@material-ui/core/ListItem";
import ListItemIcon from "@material-ui/core/ListItemIcon";

// react router dom methods
import { NavLink, withRouter } from "react-router-dom";

// list of icons used in the side navigation component
import IconButton from "@material-ui/core/IconButton";
import MenuIcon from "@material-ui/icons/Dehaze";
import ChevronLeftIcon from "@material-ui/icons/ChevronLeft";
import HomeIcon from "@material-ui/icons/Home";
import AboutIcon from "@material-ui/icons/Code";
import BillingIcon from "@material-ui/icons/Payment";
import SettingsIcon from "@material-ui/icons/Settings";
import InvoicesIcon from "@material-ui/icons/Receipt";
import { UserConsumer } from "../../contexts/UserContext";

// imported css here
import "./SideNavigation.css";

// imported components here
import AuthLanding from "../AuthLanding";
import AuthSecured from "../AuthSecured";

class SideNavigation extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      open: false,
      creditsOrPlan: ""
    };
  }

  handleDrawerOpen = () => {
    // change open state to true || false to open or close navigation
    this.setState({ open: !this.state.open });
  };

  signInModal = () => {
    // signInModal from App Component
    return this.props.signInModal();
  };

  signUpModal = () => {
    return this.props.signUpModal();
  };

  connectUserContextWithState = company => {
    console.log(company);
    if (!this.state.creditsOrPlan) {
      const { credits, unlimited_tier } = company;
      this.setState({
        creditsOrPlan: unlimited_tier ? "1 Month Unlimited" : credits
      });
    }
  };

  componentDidUpdate(prevProps) {
    if (this.props.userState !== prevProps.userState) {
      this.connectUserContextWithState(this.props.userState.companies[0]);
    }
  }

  render() {
    // deconstruct state to get a list of needed attributes
    const { open } = this.state;
    return (
      <div className="navigation">
        <header position="fixed" className="navbar">
          <IconButton
            color="inherit"
            aria-label="Open drawer"
            onClick={this.handleDrawerOpen}
          >
            <MenuIcon id="hamburger" />
          </IconButton>
          {/* if logged in show signup/signin else show */}
          {!this.props.loggedIn ? (
            <AuthLanding
              signInModal={this.signInModal}
              signUpModal={this.signUpModal}
            />
          ) : (
            <AuthSecured
              {...this.props}
              signOut={this.props.signOut}
              // credits={this.state.creditsOrPlan}
            />
          )}
        </header>
        <Drawer
          anchor={document.body.clientWidth < 520 ? "top" : "left"}
          open={open}
        >
          <div id="drawer-container">
            <IconButton onClick={this.handleDrawerOpen}>
              <ChevronLeftIcon id="left-icon" />
            </IconButton>
          </div>
          <Divider />
          {/* if not logged in show public routes else show secured */}
          {!this.props.loggedIn ? (
            <List className="all-icon-container">
              {[
                { title: "Home", icon: <HomeIcon /> },
                { title: "About", icon: <AboutIcon /> }
              ].map((text, index) => {
                const { title, icon } = text;
                return (
                  <NavLink
                    exact
                    to={`${title}` === "Home" ? "/" : `/${title}`}
                    key={title}
                    className="icon-container"
                    onClick={() => {
                      this.setState({ open: !open });
                    }}
                  >
                    <ListItem className="icon-item">
                      <ListItemIcon>{icon}</ListItemIcon>
                      <p className="icon-title">{title}</p>
                    </ListItem>
                  </NavLink>
                );
              })}
            </List>
          ) : (
            <List className="all-icon-container">
              {[
                { title: "Invoices", icon: <InvoicesIcon /> },
                { title: "Billing", icon: <BillingIcon /> },
                { title: "Settings", icon: <SettingsIcon /> }
              ].map((text, index) => {
                const { title, icon } = text;
                const lowerTitle = title.toLowerCase();
                return (
                  <NavLink
                    exact
                    to={`/user/${this.props.userState.userID}/${lowerTitle}`}
                    key={title}
                    className="icon-container"
                    onClick={() => {
                      this.setState({ open: !open });
                    }}
                  >
                    <ListItem className="icon-item">
                      <ListItemIcon>{icon}</ListItemIcon>
                      <p className="icon-title">{title}</p>
                    </ListItem>
                  </NavLink>
                );
              })}
            </List>
          )}
        </Drawer>
      </div>
    );
  }
}

export default withRouter(SideNavigation);
