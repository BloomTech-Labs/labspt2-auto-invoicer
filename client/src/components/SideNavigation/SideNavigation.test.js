import React from "react";
import { render } from "react-testing-library";
import "jest-dom/extend-expect";
import { BrowserRouter as Router } from "react-router-dom";
import SideNavigation from "../SideNavigation";

describe("<SideNavigation />", () => {
  it("renders Side Navigation component", () => {
    const { asFragment } = render(
      <Router>
        <SideNavigation />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
