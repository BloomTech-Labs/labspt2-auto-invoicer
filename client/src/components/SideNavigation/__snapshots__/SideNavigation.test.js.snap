// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideNavigation /> renders Side Navigation component 1`] = `
<DocumentFragment>
  <div
    class="navigation"
  >
    <header
      class="navbar"
      position="fixed"
    >
      <button
        aria-label="Open drawer"
        class="MuiButtonBase-root-7 MuiIconButton-root-1 MuiIconButton-colorInherit-2"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label-6"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root-10"
            focusable="false"
            id="hamburger"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M2 15.5v2h20v-2H2zm0-5v2h20v-2H2zm0-5v2h20v-2H2z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root-33"
        />
      </button>
      <ul
        class="auth-container"
      >
        <button
          class="authentication-btns"
        >
          Sign In
        </button>
      </ul>
    </header>
  </div>
</DocumentFragment>
`;
